doc/nrf/releases_and_maturity/migration/migration_guide_2.7.rst

```diff
@@ -71,6 +71,19 @@ Peripheral samples
   * The CLI command ``fem tx_power_control <tx_power_control>`` replaces ``fem tx_gain <tx_gain>`` .
     This change applies to the sample built with the :ref:`CONFIG_RADIO_TEST_POWER_CONTROL_AUTOMATIC <CONFIG_RADIO_TEST_POWER_CONTROL_AUTOMATIC>` set to ``n``.
 
+
+Matter
+------
+
+  * With the inheritance of Zephyr's :ref:`zephyr:sysbuild` in the |NCS|, some changes are provided to the Matter samples and applications:
+
+    * :kconfig:option:`CONFIG_CHIP_FACTORY_DATA_BUILD` kconfig option is deprecated and you need to use the :kconfig:option:`SB_CONFIG_MATTER_FACTORY_DATA` Kconfig option instead to enable or disable creating the factory data set during building a Matter sample.
+      To enable factory data support on your device, you still need to set the :kconfig:option:`CONFIG_CHIP_FACTORY_DATA` to ``y``.
```

Comment:
Really? A user should not have to set this, sysbuild should propagate that setting

Reply:
But one thing is to generate factory data, other to enable support for that in Matter sample. Sysbuild forced us to use this config but it is used only for generate factory data files. In fact sysbuild complicated our factory data.

Reply:
So should the SB option being set, force set the option set in the default image?

Reply:
OK, so the main thing is that we used to have two configs: CONFIG_CHIP_FACTORY_DATA that enables support for factory data and does not generate factory data set (users can do it using their scripts) and it was used just to enable parsing fd in the application. The second one: CONFIG_CHIP_FACTORY_DATA_BUILD was enabling the CMAKE script for generating factory data set. 
After moving to sysbuild only the second one became "deprecated" because you suggested creating the SB_CONFIG_MATTER_FACTORY_DATA config that only generates the factory data set. So now it is complicated, and we want to make it as easy as possible for customers and not provide many changes because they're using those mechanisms in their production. I thought that the easiest way is to inform users that CONFIG_CHIP_FACTORY_DATA_BUILD has been deprecated and now they must use SB_CONFIG_MATTER_FACTORY_DATA instead. But on the other hand, the SB_CONFIG_MATTER_FACTORY_DATA kconfig works only for sysbuild, so the old one must be available for non-sysbuild purposes, right? All other things are exactly the same, and they can manipulate the CONFIG_CHIP_FACTORY_DATA kconfig to enable/disable fd support. 

Are you seeing a better option?

Reply:
I will change this kconfig to `SB_CONFIG_MATTER_FACTORY_DATA_BUILD`, which describes the purpose better.

Reply:
does this need updating?

Reply:
Right, there is no _BUILD kconfig anymore.

Reply:
It is correct, Users need to use `SB_CONFIG_MATTER_FACTORY_DATA_GENERATE` to generate factory data, but apart from that they need to use `CONFIG_CHIP_FACTORY_DATA` to enable factory data support on SoC.